tests/core/chain-object/test_chain_retrieval_of_vm_class.py

```diff
@@ -58,24 +58,24 @@ def chaindb(base_db):
 def test_header_chain_get_vm_class_for_block_number(base_db, genesis_header):
     chain = ChainForTesting.from_genesis_header(base_db, genesis_header)
 
-    assert chain.get_vm_class_for_block_number(0) is VM_A
+    assert type(chain.get_vm_class_for_block_number(0)) is type(VM_A)
```

Comment:
This seems like a bug, or it needs more explanation.

The VM class is already a type. `type(class)` is always `type`, so this seems like a no-op check.

This whole test may have to be re-written to show that the VMs transition appropriately, even though the VMs are now modified from the originally-specified VMs.

Reply:
The thing is that `chain.get_vm_class_for_block_number(0) is VM_A` does fail because when we apply a `ConsensusEngine` we are creating a new class by calling `.configure(...)` and overwriting `self.vm_configuration`

```python
cls.vm_configuration = cls.consensus_engine.amend_vm_configuration(cls.vm_configuration)
```

Reply:
Btw, we *could* also use something like `setattr` to modify the vm class directly rather than creating new ones but I'm not sure if that's any better.

Reply:
>so this seems like a no-op check.

Oh, I just re-read your comment and now I understand! You are correct...damn...mmh. How do I compare these now :thinking:

Reply:
Ok, I'll bite the bullet and use `setattr` instead of returning new classes for these amendments. Actually `setattr` seems more in line with the word amendment anyway :sweat_smile: